### PR TITLE
Mov 12

### PR DESCRIPTION
**Pull request by Sergey113222**
1. create MovieDto
2. create SearchDto
3. create SearchMovieController (searchByName)
3.1. add searchPopular
3.2. add searchLatest
4. create test for method (searchByName & searchLatest)

### CODE REVIEW

File: src/main/java/info/movietrash/cinemabase/controller/RegistrationController.java [hunk 1]
```diff
@@ -3,6 +3,7 @@
 import info.movietrash.cinemabase.dto.UserDto;
 import info.movietrash.cinemabase.service.UserService;
 import lombok.AllArgsConstructor;
+import lombok.RequiredArgsConstructor;
```
Comment (RomanKashenok) [inline src/main/java/info/movietrash/cinemabase/controller/RegistrationController.java]: remove

File: src/main/java/info/movietrash/cinemabase/service/impl/SearchServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+    public SearchServiceImpl(RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+    }
+
+    @Override
+    public List<MovieDto> searchMoviesByName(SearchDto dto) throws JsonProcessingException {
+        URI uri = UriComponentsBuilder.newInstance()
+                .scheme("https")
+                .host("api.themoviedb.org")
```
Comment (RomanKashenok) [inline src/main/java/info/movietrash/cinemabase/service/impl/SearchServiceImpl.java]: move to properties

File: src/main/java/info/movietrash/cinemabase/service/impl/SearchServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                .toUri();
+
+        RequestEntity request = new RequestEntity(HttpMethod.GET, uri);
+        ResponseEntity<String> response = restTemplate.exchange(request, String.class);
+        String moviesJson = response.getBody();
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        JsonNode responseBody = objectMapper.readTree(moviesJson);
+        JsonNode resultsMassive = responseBody.path("results");
```
Comment (RomanKashenok) [inline src/main/java/info/movietrash/cinemabase/service/impl/SearchServiceImpl.java]: move to class constants

File: src/main/java/info/movietrash/cinemabase/service/impl/SearchServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+                .path("/3/movie/latest")
+                .queryParam("api_key", apiKey)
+                .build()
+                .toUri();
+
+        RequestEntity request = new RequestEntity(HttpMethod.GET, uri);
+        ResponseEntity<String> response = restTemplate.exchange(request, String.class);
+        String moviesJson = response.getBody();
+
+        ObjectMapper objectMapper = new ObjectMapper();
```
Comment (RomanKashenok) [inline src/main/java/info/movietrash/cinemabase/service/impl/SearchServiceImpl.java]: move to class variable

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -1,12 +1,12 @@
 spring.jpa.hibernate.ddl-auto=none
 spring.datasource.url=jdbc:mysql://localhost:3306/movie_trash?createDatabaseIfNotExist=true&allowPublicKeyRetrieval=\
   true&useSSL=false&useUnicode=true&characterEncoding=UTF-8&serverTimezone=UTC
-spring.datasource.username=
-spring.datasource.password=
+spring.datasource.username=root
+spring.datasource.password=Gr113222
```
Comment (RomanKashenok) [inline src/main/resources/application.properties]: remove from repository version

**Review by RomanKashenok — CHANGES_REQUESTED** [review]
fix comments

File: src/main/java/info/movietrash/cinemabase/service/impl/SearchServiceImpl.java (hunk 2 above)
Comment (Sergey113222) [reply src/main/java/info/movietrash/cinemabase/service/impl/SearchServiceImpl.java]: add to properties (scheme, host)

File: src/main/java/info/movietrash/cinemabase/controller/RegistrationController.java (hunk 1 above)
Comment (Sergey113222) [reply src/main/java/info/movietrash/cinemabase/controller/RegistrationController.java]: +

File: src/main/java/info/movietrash/cinemabase/service/impl/SearchServiceImpl.java (hunk 3 above)
Comment (Sergey113222) [reply src/main/java/info/movietrash/cinemabase/service/impl/SearchServiceImpl.java]: add private final String jsonNodeStr = "results"

File: src/main/java/info/movietrash/cinemabase/service/impl/SearchServiceImpl.java (hunk 4 above)
Comment (Sergey113222) [reply src/main/java/info/movietrash/cinemabase/service/impl/SearchServiceImpl.java]: add objectMapper to constructor for @Autowired

File: src/main/resources/application.properties (hunk 5 above)
Comment (Sergey113222) [reply src/main/resources/application.properties]: +

**Review by RomanKashenok — APPROVED** [review]
make json key constant like

**Pull request merged**